Sources/Sentry/Public/SentrySpan.h

```diff
@@ -1,9 +1,97 @@
-#import <Foundation/Foundation.h>
+#import "SentrySpanContext.h"
+#import <Sentry/Sentry.h>
 
 NS_ASSUME_NONNULL_BEGIN
 
+@class SentryTransaction, SentrySpanId, SentryId, SentryHub;
+
 NS_SWIFT_NAME(Span)
-@protocol SentrySpan <NSObject>
+@interface SentrySpan : SentrySpanContext
+SENTRY_NO_INIT
+
+/**
+ * The timestamp of which the span ended.
+ */
+@property (nullable, nonatomic, strong) NSDate *timestamp;
+
+/**
+ * The start time of the span.
+ */
+@property (nullable, nonatomic, strong) NSDate *startTimestamp;
+
+/**
+ * An arbitrary mapping of additional metadata of the span.
+ */
+@property (nullable, readonly) NSDictionary<NSString *, id> *extras;
+
+/**
+ * Whether the span is finished.
+ */
+@property (readonly) BOOL isFinished;
+
+/**
+ * Init a SentrySpan with given transaction, traceId, parentSpanId and hub.
+ *
+ * @param transaction The transaction associated with this span.
+ * @param traceId Determines which trace the Span belongs to.
+ * @param parentId Id of a parent span.
+ *
+ * @return SentrySpan
+ */
+- (instancetype)initWithTransaction:(SentryTransaction *)transaction
+                            traceId:(SentryId *)traceId
+                           parentId:(SentrySpanId *)parentId;
+
+/*
+ Removed because SentrySpan requires a transaction
+ */
+- (instancetype)initWithSampled:(BOOL)sampled NS_UNAVAILABLE;
+
+/*
+ Removed because SentrySpan requires a transaction
+ */
+- (instancetype)initWithTraceId:(SentryId *)traceId
+                         spanId:(SentrySpanId *)spanId
+                       parentId:(nullable SentrySpanId *)parentId
+                        sampled:(BOOL)sampled NS_UNAVAILABLE;
+
+/**
+ * Starts a child span.
+ *
+ * @param operation Defines the child span operation.
+ *
+ * @return SentrySpan
+ */
+- (SentrySpan *)startChildWithOperation:(NSString *)operation NS_SWIFT_NAME(startChild(operation:));
+
+/**
+ * Starts a child span.
+ *
+ * @param operation Defines the child span operation.
+ * @param description Define the child span description.
+ *
+ * @return SentrySpan
+ */
+- (SentrySpan *)startChildWithOperation:(NSString *)operation
+                            description:(nullable NSString *)description
```

Comment:
Thats depend. This method could be called from other method that load these values from somewhere else, and description is not available. Instead of having to check if description is null or not and call the appropriate method, the null description can be easily passed forward.

Reply:
Then I would instead create an internal constructor that accepts a `nullable` description to pass the description forward. Still, I think the public API doesn't need a `nullable` description.

Reply:
What I meant was that the user may want to create a SpanContext, and maybe he may or may not have a description, so he will have to use an if to know which method to use.

Reply:
yeah so if the description could come from somewhere, as a nullable string, and the user would need to check for null to call one method or the other. That makes sense.

var desc = getDescription();
var op = getOperation();

if (desc is null) {
  startChild(operation: op)
else {
  startChild(operation: op, description: desc)
}


Given the field is nullable anyway, it makes sense to me the overload takes a nullable string instead. If I understood @brustolin correctly here

Reply:
yep I agree because of the simplicity, what we need to think here is that some info like `name`, `op`, or `description` could come at runtime from external resources and not hardcoded values that you know what's about.
so overloads must accept nullable values (if the field is nullable of course).